lib/tapioca/helpers/rbi_helper.rb

```diff
@@ -107,36 +107,22 @@ def as_non_nilable_type(type)
 
     sig { params(name: String).returns(T::Boolean) }
     def valid_method_name?(name)
-      # try to parse a method definition with this name
-      iseq = RubyVM::InstructionSequence.compile("def #{name}; end", nil, nil, 0, false)
-      # pull out the first operation in the instruction sequence and its first argument
-      op, arg, _data = iseq.to_a.dig(-1, 0)
-      # make sure that the operation is a method definition and the method that was
-      # defined has the expected name, for example, for `def !foo; end` we don't get
-      # a syntax error but instead get a method defined as `"foo"`
-      op == :definemethod && arg == name.to_sym
-    rescue SyntaxError
-      false
+      # Special case: Prism supports parsing `def @foo; end`, but the Sorbet parser doesn't. This condition can go away
+      # once Sorbet is using Prism under the hood as it will no longer result in an RBI that Sorbet can't parse
+      return false if name.start_with?("@")
+
+      result = Prism.parse("def #{name}(a); end")
```

Comment:
Why not test `Prism.parse("def self.#{name}(a); end")` instead? I think it would address both the `@foo` concern and the `self.foo` one with just the parse.

Reply:
Oh, great idea https://github.com/Shopify/tapioca/pull/2062.